### ISSUE TITLE
doubt of  time  for  the  training  wav2letter's  960hour libri data

### ISSUE THREAD

**Issue by bolt163** [facebookresearch/wav2letter#57]
while the paper do not give a releated reference of  time for the training process,  in the paper ——"Letter-Based Speech Recognition with Gated ConvNets",  there is a representation  about the time consuming about the test data,  in  3.2 Comparison with other systems
"Using a single GPU (and no utterance batching), our High Dropout Gated ConvNet
goes over the clean (5:4h) and other (5:1h) test sets in 4min26s and 4min43s"

while I ran the traing process with the host configuration——"4  Nvidia Tesla K40M GPU, 12GB(gpu memory storage)",  for each  training epoch, it need 4 days, and for each epoch testing with the same test dataset, my host seems to consuming  8 more minutes,  which is double of the 4min plus in the paper
so i have 2 doubt about the training process,
    1, this means the host's performance is half of the  paper's relating running host, forgive me for the  speculation!  

![00](https://user-images.githubusercontent.com/25170435/36708207-eb1ce5ec-1bac-11e8-9d3a-63ff3f6474f5.png)

![01](https://user-images.githubusercontent.com/25170435/36708246-1e7baa04-1bad-11e8-9e30-a1a07b0e2869.png)

   2.  while in the  paper above, there was no presentation about  training time for each epoch??  while i am training, for each training  epoch, it consume 4 days+,   and in the  paper,  the best result for the  experiments need training 40 epoch, , so.....it seems so much slow for me if  i continue with the current  situation....
     Is there any solution for  accelerating the training process,  with better performance of GPU host? 
 multi-gpu? scaling down the training data-set?

**Comment by vineelpratap** [comment]
Hi, 
Regarding training time, we used MPI for training and I'm guessing you are not using that option. This might be the reason you are seeing relatively high training times. You can use mpi option  (try 4 processes)  specified here https://github.com/facebookresearch/wav2letter/blob/master/train.lua#L33 .

**Comment by bolt163** [comment]
yesterday I  also tried MPI  training on 3 GPU【my host have 4 GPU(Nvidia Tesla K40M,with 12G memory),but there are also other deep learning jobs, so i can't occupy all the gpu resources】

but the trainging process it seems not that fast(no exceeding my expectations), for  epoch training , it still need 2 days more
![_15197889586509](https://user-images.githubusercontent.com/25170435/36769067-07dd5b50-1c7d-11e8-9b26-ad0279ddbd89.png)
![_15197890483114](https://user-images.githubusercontent.com/25170435/36769069-0bff15d4-1c7d-11e8-9bbd-0cb5c2ba75b5.png)

**Comment by bolt163** [comment]
hi  vineelpratap, 
thanks for your detail reply, I will tray latter soon

"Hi,
Regarding training time, we used MPI for training and I'm guessing you are not using that option. This might be the reason you are seeing relatively high training times. You can use mpi option specified here https://github.com/facebookresearch/wav2letter/blob/master/train.lua#L33 . Feel free to experiment with the number of processes or use 4. 
"

**Comment by vineelpratap** [comment]
Closing this. Lets discuss on #60.
